rollbar-web/src/main/java/com/rollbar/web/filter/RollbarFilter.java

```diff
@@ -0,0 +1,71 @@
+package com.rollbar.web.filter;
+
+import static com.rollbar.notifier.config.ConfigBuilder.withAccessToken;
+
+import com.rollbar.notifier.Rollbar;
+import com.rollbar.web.provider.PersonProvider;
+import com.rollbar.web.provider.RequestProvider;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+public class RollbarFilter implements Filter {
+
+  static final String ACCESS_TOKEN_PARAM_NAME = "access_token";
+
+  static final String USER_IP_HEADER_PARAM_NAME = "user_ip_header";
+
+  private Rollbar rollbar;
```

Comment:
From my googling around, Filters are reused across threads, so this instance variable will be used across threads and therefore this code is not thread safe unless the Rollbar class is.

Reply:
Yes, there is just one instance of a filter per application context. The point is, as I mentioned, that the Rollbar instance should be thread safe, and so far I think it is if the providers are thread safe.